api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -0,0 +1,500 @@
+static void
+ml_single_set_inout_tensors_info (GObject *object,
+    const gchar *prefix, ml_tensors_info_s *tensors_info)
+{
+  GstTensorsInfo info;
+  gchar *str_dim, *str_type, *str_name;
+  gchar *str_type_name, *str_name_name;
+
+  ml_tensors_info_copy_from_ml (&info, tensors_info);
+
+  /* Set input option */
+  str_dim = gst_tensors_info_get_dimensions_string (&info);
+  str_type = gst_tensors_info_get_types_string (&info);
+  str_name = gst_tensors_info_get_names_string (&info);
+
+  str_type_name = g_strdup_printf("%s%s", prefix, "type");
+  str_name_name = g_strdup_printf("%s%s", prefix, "name");
+
+  g_object_set (object, prefix, str_dim,
+      str_type_name, str_type,
+      str_name_name, str_name, NULL);
+
+  g_free (str_type_name);
+  g_free (str_name_name);
+  g_free (str_dim);
+  g_free (str_type);
+  g_free (str_name);
+
+  gst_tensors_info_free (&info);
+}
+
+/**
+ * @brief Check the availability of the nnfw type and model
+ */
+static int
+ml_single_check_nnfw (const char *model, ml_nnfw_type_e *nnfw)
+{
+  gchar *path_down;
+  int status = ML_ERROR_NONE;
+
+  /* Check file extention. */
+  path_down = g_ascii_strdown (model, -1);
+
+  switch (*nnfw) {
+    case ML_NNFW_TYPE_ANY:
+      if (g_str_has_suffix (path_down, ".tflite")) {
+        ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
+            model);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+      } else if (g_str_has_suffix (path_down, ".pb")) {
+        ml_logi ("The given model [%s] is supposed a tensorflow model.", model);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW;
+      } else if (!g_str_has_suffix (path_down, ".so")) {
+        ml_logi ("The given model [%s] is supposed a custom filter model.",
+            model);
+        *nnfw = ML_NNFW_TYPE_CUSTOM_FILTER;
+      } else {
+        ml_loge ("The given model [%s] has unknown extension.", model);
+        status = ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    case ML_NNFW_TYPE_CUSTOM_FILTER:
+      if (!g_str_has_suffix (path_down, ".so")) {
+        ml_loge ("The given model [%s] has invalid extension.", model);
+        status = ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    case ML_NNFW_TYPE_TENSORFLOW_LITE:
+      if (!g_str_has_suffix (path_down, ".tflite")) {
+        ml_loge ("The given model [%s] has invalid extension.", model);
+        status = ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    case ML_NNFW_TYPE_TENSORFLOW:
+      if (!g_str_has_suffix (path_down, ".pb")) {
+        ml_loge ("The given model [%s] has invalid extension.", model);
+        status = ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    case ML_NNFW_TYPE_NNFW:
+      /** @todo Need to check method for NNFW */
+      ml_loge ("NNFW is not supported.");
+      status = ML_ERROR_NOT_SUPPORTED;
+      break;
+    default:
+      break;
+  }
+
+  g_free (path_down);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  if (!g_file_test (model, G_FILE_TEST_IS_REGULAR)) {
```

Comment:
I found that this can be moved to line 86, before checking the file extension. (added a PR to update this for old single-shot function.)
Plz consider to move this block.

Reply:
Yes, you are right
Updated.